adam/situation/templates/phase1_templates.py

```diff
@@ -403,30 +422,73 @@ def _instantiate_objects(
         self,
         template: Phase1SituationTemplate,
         variable_assignment: "TemplateVariableAssignment",
+        *,
+        default_addressee_node: OntologyNode,
     ):
-        object_var_to_instantiations: Mapping[
-            TemplateObjectVariable, SituationObject
-        ] = immutabledict(
-            (obj_var, self._instantiate_object(obj_var, variable_assignment))
-            for obj_var in template.all_object_variables
+        has_addressee = any(
+            IS_ADDRESSEE in object_.asserted_properties
+            for object_ in template.all_object_variables
         )
-        return object_var_to_instantiations
+
+        object_var_to_instantiations_mutable: List[
+            Tuple[TemplateObjectVariable, SituationObject]
+        ] = [
+            (
+                obj_var,
+                self._instantiate_object(
+                    obj_var,
+                    variable_assignment,
+                    has_addressee=has_addressee,
+                    default_addressee_node=default_addressee_node,
+                ),
+            )
+            for obj_var in template.all_object_variables
+        ]
+        if (
```

Comment:
Will these assign the addressee to the learner if the learner is present but not yet marked as the addressee?

Reply:
Yes. There is a unit test for this case to verify as well.